scipy/sparse/base.py

```diff
@@ -675,23 +675,75 @@ def toarray(self, order=None, out=None):
         """
         return self.tocoo().toarray(order=order, out=out)
 
-    def todok(self):
-        return self.tocoo().todok()
+    # In principle, providing a tocsr method will be sufficient for any
+    # sparse matrix format deriving from spmatrix.
+    def tocsr(copy=False):
```

Comment:
I think see why I didn't catch it: I was only running the `scipy.sparse` tests (i.e. `runtests.py -v scipy.sparse`) which didn't run the pyflakes test. However, I just ran the full suite and that doesn't seem to catch the pyflake failures either. Am I missing something here?

Also, none of the tests would catch the `.tocsr` bug, because all the matrixes override it.